providers.d/apt-mirror/provider_wrapper.sh

```diff
@@ -1,13 +1,42 @@
 bashelliteProviderWrapper() {
 
-  if [[ ! ${_r_dryrun} ]]; then
-    apt-mirror ${_r_metadata_tld}/repos.conf.d/${_n_repo_name}/provider.conf;
+  local basepath_parameter=$(egrep "^\s*set\s*base_path\s*[[:alnum:]/_-.]*" "${_r_metadata_tld}/repos.conf.d/${_n_repo_name}/provider.conf")
+  basepath_parameter="${basepath_parameter#*set*base_path}"
+  if [[  "${basepath_parameter}" != "${_r_mirror_tld}/${_n_repo_name}" ]]; then
+    utilMsg WARN "$(utilTime)" "The \"base_path\" parameter ("${basepath_parameter}") in provider.conf does not match mirror location (${_r_mirror_tld}/${_n_repo_name})..."
   fi
 
-  if [[ "${?}" != "0" ]]; then
-    utilMsg WARN "$(utilTime)" "apt-mirror either failed or completed with errors for repo (${_n_repo_name})."
+  local config_file="${_r_metadata_tld}/repos.conf.d/${_n_repo_name}/provider.conf"
+  local tmp_config_file="${HOME}/.bashellite/tmp_${_n_repo_name}_provider.conf"
+
+  > ${tmp_config_file}
+
+  local line_counter=0
+  while IFS=$'\n' read line; do
+    if [[ "${line}" =~ ^set[[:blank:]]base_path[[:blank:]]+[[:alnum:]/_-.]+ ]]; then
+      if [ "${line_counter}" = 0 ]; then
+        echo "set base_path         ${_r_mirror_tld}/${_n_repo_name}" >> ${tmp_config_file}
+      else
+        utilMsg WARN "$(utilTime)" "Duplicate base_path parameter found.  Skipping line..."
+      fi
+    else
+      echo "${line}" >> ${tmp_config_file}
+    fi
+    line_counter=$(($line_counter + 1))
```

Comment:
Same thing as bandersnatch provider wrapper. This line_counter increment needs to move up.

Reply:
Updated.